protocol/x/assets/keeper/asset.go

```diff
@@ -240,34 +229,37 @@ func (k Keeper) GetDenomById(
 func (k Keeper) GetAsset(
 	ctx sdk.Context,
 	id uint32,
-) (val types.Asset, err error) {
+) (val types.Asset, exists bool) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.AssetKeyPrefix))
 
 	b := store.Get(lib.Uint32ToBytes(id))
 	if b == nil {
-		return val, errorsmod.Wrap(types.ErrAssetDoesNotExist, lib.Uint32ToString(id))
+		return val, false
 	}
 
 	k.cdc.MustUnmarshal(b, &val)
-	return val, nil
+	return val, true
 }
 
 func (k Keeper) GetAllAssets(
 	ctx sdk.Context,
-) []types.Asset {
-	num := k.GetNumAssets(ctx)
-	assets := make([]types.Asset, num)
-
-	for i := uint32(0); i < num; i++ {
-		asset, err := k.GetAsset(ctx, i)
-		if err != nil {
-			panic(err)
-		}
+) (list []types.Asset) {
```

Comment:
Given that this builds on top of [this PR](https://github.com/dydxprotocol/v4-chain/pull/414), should we label this as state-breaking too (since we are introducing the iterator here)? Or are we assuming that labelling the above PR is sufficient?

Reply:
I think it's worth clarifying what `state-breaking` means.
IMO this PR is `state-breaking` not only due to the iterator, but also due to the fact that we added additional validatoin on `CreateAsset` ("state-incompatible")